openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -235,47 +211,76 @@ def get_cohort(user, course_key, assign=True):
         if not assign:
             return None
 
-    choices = course.auto_cohort_groups
-    if len(choices) > 0:
-        # Randomly choose one of the auto_cohort_groups, creating it if needed.
-        group_name = local_random().choice(choices)
+    course_cohorts = get_course_cohorts(course, assignment_type=CourseUserGroup.RANDOM)
+    if course_cohorts:
+        course_cohort = local_random().choice(course_cohorts)
     else:
-        # Use the "default cohort".
-        group_name = DEFAULT_COHORT_NAME
+        course_cohort, __ = CourseUserGroup.objects.get_or_create(
+            course_id=course_key,
+            group_type=CourseUserGroup.COHORT,
+            name=DEFAULT_COHORT_NAME,
+            defaults={'assignment_type': CourseUserGroup.RANDOM}
+        )
 
-    group, __ = CourseUserGroup.objects.get_or_create(
-        course_id=course_key,
-        group_type=CourseUserGroup.COHORT,
-        name=group_name
+    user.course_groups.add(course_cohort)
+
+    return course_cohort
+
+
+def migrate_cohort_settings(course):
```

Comment:
@explorerleslie note also that there is a little bit of weirdness that the discussion topics ARE exported with the course, the topics might be specifically made for cohorts, which are not going to be exported. Of course, the long-term dream is to move discussion topics out of Studio also.